aisimulate/src/aisimulate/sweeper/search.py

```
@@ -142,27 +406,123 @@ def _evaluate_one(
+        effective_targets.intersection(
+            {OptimizationTarget.GOODPUT, OptimizationTarget.GOODPUT_PER_GPU}
+        )
+        and "goodput_output_throughput_tok_s" not in report
+    ):
+        return (
+            None,
+            None,
+            "failed",
+            "runner contract violation: goodput objective requires "
+            "goodput_output_throughput_tok_s; aggregate latency cannot be used "
+            "as a fallback",
```

Ruff (0.16.0)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)